src/transformers/generation_utils.py

```diff
@@ -545,8 +546,12 @@ def _generate_no_beam_search(
             )
 
             # if model has past, then set the past variable to speed up decoding
-            if self._use_cache(outputs, use_cache):
-                past = outputs[1]
+            if outputs.get("past_key_values", None) is not None:
```

Comment:
We control the output type now, can we control the cache key? 
Maybe Seq2SeqLMOutput.decoder_past_key_values -> past_key_values, since there are no encoder_past_key_values afaik?

Reply:
Yeah, I see what you mean! 
I guess the trade-off is:
a) If we leave it at `decoder_past_key_values`, then we have two different names for essentially the same thing in the generate function `decoder_past_key_values` and `past_key_values`
b) If we change it to `past_key_values`, then people might think that it corresponds to the encoder part of the encoder-decoder (similar to attention_mask, input_ids), but on the other hand variables like `labels` are also not called `decoder_labels` as it is pretty clear that this param corresponds to the decoder only. 

So, I would be fine with changing all `decoder_past_key_values` to `past_key_values` (leaving the alias for backward compatibility), but not 100% sure if it's worth it. 
Is that what you meant ? @sshleifer 
Also would be cool to get your opinions on that @LysandreJik @sgugger

Reply:
yes that's what I meant. We still haven't had a release since we introduced ModelOutputs, right?

Reply:
+1 for `past_key_values`: I'm not too concerned about people thinking they're an encoder variable, and I think it goes in the right direction of unifying causal LM / seq2seq generation.

Reply:
Cool did the renaming